tools: clean installer and config tool

lahm86 opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This is a housekeeping update to clear all warnings and messages from the two C# projects (example [here](https://github.com/LostArtefacts/TR1X/actions/runs/6247127192)). I've left the commits separate to mark clearly what was done, but will squash after review.

The one extended change is in replacing `WebClient` for file downloads. The class is now obsolete and its replacement, `HttpClient`, doesn't have a progress tracker; I've created a wrapper for that so there is no loss of functionality.

lahm86 committed:
tools: generate assembly info

Removes warnings about using Windows-only modules such as
Microsoft.Win32.Registry.

lahm86 committed:
tools: fix naming violations

lahm86 committed:
installer: clear misc messages

Simplified new statements and null checks; used tuple swapping; made
some fields readonly; used explicit casting where necessary.

lahm86 committed:
installer: update method signatures

Made some methods static, removed an unused method and created local
functions.

lahm86 committed:
installer: replace obsolete WebClient

Replaced using the obsolete WebClient with a custom HttpClient that
supports progress tracking.

lahm86 committed:
installer: always use UI thread for messages

Ensures that messages generated during the install process are logged
via the UI thread, and sets the UI message box as readonly.

lahm86 committed:
config: restore EOF newlines

rr- commented:
Hey, in the naming violations commit we only change EventHandler types but leave other private variables intact.  In my opinion this change introduces inconsistency. Is there somewhere I can read more about this warning?

lahm86 committed:
tools: fix relay command event naming

lahm86 commented:
> Hey, in the naming violations commit we only change EventHandler types but leave other private variables intact. In my opinion this change introduces inconsistency. Is there somewhere I can read more about this warning?

I didn't think this through properly yesterday. I've now removed the `event` keyword as that's not required in these cases, and reverted the naming. The `event` keyword does the following, which is why VS complains about the naming i.e. it's similar to property naming.

```cs
private event EventHandler Temp;
```
=>
```cs
private EventHandler _temp;
private event EventHandler Temp
{
    add
    {
        _temp += value;
    }
    remove
    {
        _temp -= value;
    }
}
```

But in the `RelayCommand` case, we are adding to/removing from the handler directly via, for example, `public event EventHandler? CanExecuteChanged`. So having the additional `event` keyword on the private variable was redundant.
